### PR TITLE
Copy TabChild::HandlePossibleViewportChange() to TabChildHelper

### PR DESCRIPTION
**Pull request by rojkov**

### CODE REVIEW

File: embedding/embedlite/utils/TabChildHelper.cpp [hunk 1]
```diff
@@ -226,33 +300,144 @@ class EmbedUnloadScriptEvent : public nsRunnable
...
+{
   LOGF();
   mozilla::CSSRect cssCompositedRect = aFrameMetrics.CalculateCompositedRectInCssPixels();
   // The BrowserElementScrolling helper must know about these updated metrics
   // for other functions it performs, such as double tap handling.
-
-  nsCOMPtr<nsPIDOMWindow> window = do_GetInterface(mView->mWebNavigation);
-  nsCOMPtr<nsIDOMWindowUtils> utils = do_GetInterface(window);
-
+  // TODO: send this JSON conditionally if sPostAZPCAsJson.viewport is true (embedlite.azpc.json.viewport)
```
Comment (tmeshkova) [inline embedding/embedlite/utils/TabChildHelper.cpp]: Before we had this message fired only by pref enabled 
if (sPostAZPCAsJson.viewport) { 
but now it is going to be fired all the time regardless embed client needs.

**Comment by tmeshkova** [comment]
Please add preference which would allow to enable/disable this viewport handler inside embedlite.. so Embed client could control it.

File: embedding/embedlite/utils/TabChildHelper.cpp [hunk 2]
```diff
@@ -226,33 +300,144 @@ class EmbedUnloadScriptEvent : public nsRunnable
...
+TabChildHelper::ProcessUpdateSubframe(nsIContent* aContent,
+                                const FrameMetrics& aMetrics)
+{
+  // scroll the frame to the desired spot
+  nsIScrollableFrame* scrollFrame = nsLayoutUtils::FindScrollableFrameFor(aMetrics.mScrollId);
+  if (scrollFrame) {
+    scrollFrame->ScrollToCSSPixelsApproximate(aMetrics.mScrollOffset);
+  }
+
+  nsCOMPtr<nsIDOMWindowUtils> utils(::GetDOMWindowUtils(aContent));
```
Comment (tmeshkova) [inline embedding/embedlite/utils/TabChildHelper.cpp:441]: do you really need :: here ? it is local function IIUC

File: embedding/embedlite/utils/TabChildHelper.cpp [hunk 3]
```diff
@@ -602,3 +787,169 @@ class EmbedUnloadScriptEvent : public nsRunnable
...
+{
+  nsCOMPtr<nsIDOMDocument> domDoc;
+  mView->mWebNavigation->GetDocument(getter_AddRefs(domDoc));
+  nsCOMPtr<nsIDocument> document(do_QueryInterface(domDoc));
+
+  nsCOMPtr<nsIDOMWindowUtils> utils(GetDOMWindowUtils());
+
+  nsViewportInfo viewportInfo = nsContentUtils::GetViewportInfo(document, mInnerSize);
+  mView->SendUpdateZoomConstraints(viewportInfo.IsZoomAllowed(),
+                            viewportInfo.GetMinZoom().scale,
```
Comment (tmeshkova) [inline embedding/embedlite/utils/TabChildHelper.cpp]: nit, align indent here plz

**Comment by tmeshkova** [comment]
Looks good with comments fixed

File: embedding/embedlite/utils/TabChildHelper.cpp (hunk 2 above)
Comment (rojkov) [inline embedding/embedlite/utils/TabChildHelper.cpp:441]: Actually there are two local functions with the same name, but different arguments. They are both used in the patch. There are two reasons why I copied them:
1. I decided to stick to the original TabChild code as much as possible, so future changes in TabChild could be easily mapped to TabChildHelper
2. the functions actually make the code cleaner

Comment (tmeshkova) [inline embedding/embedlite/utils/TabChildHelper.cpp:441]: ok, make sense

**Pull request merged**